store/sqlstore/channel_store.go

```diff
@@ -2854,6 +2855,30 @@ func (s SqlChannelStore) channelSearchQuery(term string, opts store.ChannelSearc
 		query = query.Where("c.Id NOT IN (SELECT ChannelId FROM GroupChannels WHERE GroupChannels.GroupId = ? AND GroupChannels.DeleteAt = 0)", opts.NotAssociatedToGroup)
 	}
 
+	if len(opts.TeamIds) > 0 {
+		query = query.Where(sq.Eq{"c.TeamId": opts.TeamIds})
+	}
+
+	if opts.GroupConstrained {
+		query = query.Where(sq.Eq{"c.GroupConstrained": true})
+	} else if opts.ExcludeGroupConstrained {
+		query = query.Where(sq.Or{
+			sq.NotEq{"c.GroupConstrained": true},
+			sq.Eq{"c.GroupConstrained": nil},
```

Comment:
nit: this is not redundant? I mean, if the `GroupConstrained` is `true`, it is not `nil`, right?

Reply:
Unfortunately it doesnt work that way... if a column has a `NULL` value then it needs to be compared with `NULL`, so a column that has a value of `NULL` doesnt get returned if you do `Column <> true `